tests/std/tests/P0645R10_text_formatting_grapheme_clusterization/test.cpp

```diff
@@ -0,0 +1,416 @@
+    {{U'\x378', U'\x308', U'\x1f1e6'}, {0ull, 2ull, 3ull}}, {{U'\x378', U'\x600'}, {0ull, 1ull, 2ull}},
+    {{U'\x378', U'\x308', U'\x600'}, {0ull, 2ull, 3ull}}, {{U'\x378', U'\x903'}, {0ull, 2ull}},
+    {{U'\x378', U'\x308', U'\x903'}, {0ull, 3ull}}, {{U'\x378', U'\x1100'}, {0ull, 1ull, 2ull}},
+    {{U'\x378', U'\x308', U'\x1100'}, {0ull, 2ull, 3ull}}, {{U'\x378', U'\x1160'}, {0ull, 1ull, 2ull}},
+    {{U'\x378', U'\x308', U'\x1160'}, {0ull, 2ull, 3ull}}, {{U'\x378', U'\x11a8'}, {0ull, 1ull, 2ull}},
+    {{U'\x378', U'\x308', U'\x11a8'}, {0ull, 2ull, 3ull}}, {{U'\x378', U'\xac00'}, {0ull, 1ull, 2ull}},
+    {{U'\x378', U'\x308', U'\xac00'}, {0ull, 2ull, 3ull}}, {{U'\x378', U'\xac01'}, {0ull, 1ull, 2ull}},
+    {{U'\x378', U'\x308', U'\xac01'}, {0ull, 2ull, 3ull}}, {{U'\x378', U'\x231a'}, {0ull, 1ull, 2ull}},
+    {{U'\x378', U'\x308', U'\x231a'}, {0ull, 2ull, 3ull}}, {{U'\x378', U'\x300'}, {0ull, 2ull}},
+    {{U'\x378', U'\x308', U'\x300'}, {0ull, 3ull}}, {{U'\x378', U'\x200d'}, {0ull, 2ull}},
+    {{U'\x378', U'\x308', U'\x200d'}, {0ull, 3ull}}, {{U'\x378', U'\x378'}, {0ull, 1ull, 2ull}},
+    {{U'\x378', U'\x308', U'\x378'}, {0ull, 2ull, 3ull}},
+    {{U'\xd', U'\xa', U'\x61', U'\xa', U'\x308'}, {0ull, 2ull, 3ull, 4ull, 5ull}}, {{U'\x61', U'\x308'}, {0ull, 2ull}},
+    {{U'\x20', U'\x200d', U'\x646'}, {0ull, 2ull, 3ull}}, {{U'\x646', U'\x200d', U'\x20'}, {0ull, 2ull, 3ull}},
+    {{U'\x1100', U'\x1100'}, {0ull, 2ull}}, {{U'\xac00', U'\x11a8', U'\x1100'}, {0ull, 2ull, 3ull}},
+    {{U'\xac01', U'\x11a8', U'\x1100'}, {0ull, 2ull, 3ull}},
+    {{U'\x1f1e6', U'\x1f1e7', U'\x1f1e8', U'\x62'}, {0ull, 2ull, 3ull, 4ull}},
+    {{U'\x61', U'\x1f1e6', U'\x1f1e7', U'\x1f1e8', U'\x62'}, {0ull, 1ull, 3ull, 4ull, 5ull}},
+    {{U'\x61', U'\x1f1e6', U'\x1f1e7', U'\x200d', U'\x1f1e8', U'\x62'}, {0ull, 1ull, 4ull, 5ull, 6ull}},
+    {{U'\x61', U'\x1f1e6', U'\x200d', U'\x1f1e7', U'\x1f1e8', U'\x62'}, {0ull, 1ull, 3ull, 5ull, 6ull}},
+    {{U'\x61', U'\x1f1e6', U'\x1f1e7', U'\x1f1e8', U'\x1f1e9', U'\x62'}, {0ull, 1ull, 3ull, 5ull, 6ull}},
+    {{U'\x61', U'\x200d'}, {0ull, 2ull}}, {{U'\x61', U'\x308', U'\x62'}, {0ull, 2ull, 3ull}},
+    {{U'\x61', U'\x903', U'\x62'}, {0ull, 2ull, 3ull}}, {{U'\x61', U'\x600', U'\x62'}, {0ull, 1ull, 3ull}},
+    {{U'\x1f476', U'\x1f3ff', U'\x1f476'}, {0ull, 2ull, 3ull}}, {{U'\x61', U'\x1f3ff', U'\x1f476'}, {0ull, 2ull, 3ull}},
+    {{U'\x61', U'\x1f3ff', U'\x1f476', U'\x200d', U'\x1f6d1'}, {0ull, 2ull, 5ull}},
+    {{U'\x1f476', U'\x1f3ff', U'\x308', U'\x200d', U'\x1f476', U'\x1f3ff'}, {0ull, 6ull}},
+    {{U'\x1f6d1', U'\x200d', U'\x1f6d1'}, {0ull, 3ull}}, {{U'\x61', U'\x200d', U'\x1f6d1'}, {0ull, 2ull, 3ull}},
+    {{U'\x2701', U'\x200d', U'\x2701'}, {0ull, 3ull}}, {{U'\x61', U'\x200d', U'\x2701'}, {0ull, 2ull, 3ull}}};
+
+
+bool run_unicode_test_data_utf32() {
+    for (size_t test_idx = 0; test_idx < size(test_data); ++test_idx) {

```

Comment:
Why not use a range based for?

Reply:
the actual reason is that during debugging I wanted to break on a particular index. Now that debugging is complete I can change this back to a range-based for